Assets/Plugins/Editor/Uplift/Schemas/Repositories/GithubRepository.cs

```diff
@@ -0,0 +1,155 @@
+            TemporaryDirectory td = new TemporaryDirectory();
+            using(StreamReader sr = new StreamReader(GitHub.GetAssetStream(packageAsset, GetToken())))
+            {
+                UnityPackage unityPackage = new UnityPackage();
+                unityPackage.Extract(sr.BaseStream, td.Path);
+            }
+            return td;
+        }
+
+        public override Upset[] ListPackages()
+        {
+            releases = GetPackagesReleases();
+
+            GitHubAsset[] upsetAssets = releases
+                .SelectMany<GitHubRelease, GitHubAsset>(rel => rel.assets.Where(asset => asset.name.EndsWith("Upset.xml")))
+                .ToArray();
+
+            string progressBarTitle = "Parsing Upsets from GitHub repository";
+            int index = 0;
+
+            List<Upset> upsetList = new List<Upset>();
+            EditorUtility.DisplayProgressBar(progressBarTitle, "Please wait a little bit while Uplift parses the Upset in the GitHub repository at " + urlField, 0f);
+            foreach(GitHubAsset asset in upsetAssets)
+            {
+                StrictXmlDeserializer<Upset> deserializer = new StrictXmlDeserializer<Upset>();
+                
+                EditorUtility.DisplayProgressBar(
+                    progressBarTitle,
+                    "Parsing " + asset.name, 
+                    (float)(index++) / upsetAssets.Length
+                );
+                
+                using(StreamReader sr = new StreamReader(GitHub.GetAssetStream(asset, GetToken())))
+                {
+                    Upset upset = deserializer.Deserialize(sr.BaseStream);
+                    upset.MetaInformation.dirName = asset.name;
+                    upsetList.Add(upset);
+                }
+            }
+
+            EditorUtility.ClearProgressBar();
+            return upsetList.ToArray();
+        }
+
+        private GitHubRelease[] GetPackagesReleases()
+        {
+            if(releases == null)
+            {
+                IEnumerator e = GitHub.LoadReleases(urlField, GetToken());
+                do { Thread.Sleep(1000); } while (e.MoveNext());
+
+                GitHubRelease[] fetchedReleases = (GitHubRelease[]) e.Current;
+
+                if (fetchedReleases == null)
+                    throw new ApplicationException("This github repository does not have releases");
+
+                string[] tagArray;
+                if(tagListField == null || tagListField.Length == 0)
+                {
+                    tagArray = new string[] { "packages" };
+                }
+                else
+                {
+                    tagArray = tagListField;
+                    Array.Resize<string>(ref tagArray, tagListField.Length + 1);
+                    tagArray[tagListField.Length] = "packages";
+                }
+
+                releases = fetchedReleases.Where(rel => tagArray.Contains(rel.tag)).ToArray();
```

Comment:
would it be better to fail if the specified repository aren't found? e.g. misconfigured.

In which case we probably don't want to add "packages" if it isn't specified.